xarray/tests/test_backends.py

```diff
@@ -1404,6 +1404,24 @@ def test_refresh_from_disk(self) -> None:
                     a.close()
                     b.close()
 
+    def test_byte_attrs(self) -> None:
+        with create_tmp_file() as tmp_file:
+            try:
+                null_byte = b"\x00"
+                other_bytes = bytes(range(1, 256))
+                ds = Dataset({"x": 1}, coords={"x_coord": [1]})
+                ds["x"].attrs["null_byte"] = null_byte
+                ds["x"].attrs["other_bytes"] = other_bytes
+                self.save(ds, tmp_file)
+            except ValueError:
+                assert self.engine == "h5netcdf"
+            else:
+                with self.open(tmp_file) as ds_out:
+                    assert ds_out["x"].attrs["null_byte"] == ""
+                    assert ds_out["x"].attrs["other_bytes"] == other_bytes.decode(
+                        errors="replace"
+                    )
+
```

Comment:
One way to clean this up would be to rewrite this to
```python
    def test_byte_attrs(self, byte_attrs_dataset) -> None:
        # TODO: consider using self.roundtrip
        with create_tmp_file() as tmp_file:
            self.save(ds, tmp_file)
            with self.open(tmp_file) as ds_out:
                assert ds_out["x"].attrs["null_byte"] == ""
                assert ds_out["x"].attrs["other_bytes"] == other_bytes.decode(
                    errors="replace"
                )

```

and in the h5netcdf tests we could override
```python
def test_byte_attrs(self, byte_attrs_dataset):
    with pytest.raises(ValueError, ADD_REGEX_HERE):
        super().test_byte_attrs(dataset)
```

and then elsewhere (`conftest.py`) create the dataset as a fixture

```python
@pytest.fixture
def byte_attrs_dataset():
null_byte = b"\x00"
    other_bytes = bytes(range(1, 256))
    ds = Dataset({"x": 1}, coords={"x_coord": [1]})
    ds["x"].attrs["null_byte"] = null_byte
    ds["x"].attrs["other_bytes"] = other_bytes
	return ds
```

Reply:
Awesome, I've made these changes! Really appreciate the feedback.

Reply:
Nice work. Thanks!